test/lunatech/lunchplanner/persistence/MenuPerDayTableSpec.scala

```diff
@@ -1,110 +1,200 @@
+import org.scalacheck.Prop.forAll
+import org.scalacheck.{ Gen, Properties }
 
 import scala.concurrent.Await
-import scala.concurrent.duration._
-
-class MenuPerDayTableSpec extends AcceptanceSpec with TestDatabaseProvider {
-  implicit private val dbConnection = app.injector.instanceOf[DBConnection]
-
-  private val newMenu = Menu(name = "Main menu")
-  private val newMenuPerDay = MenuPerDay(menuUuid = newMenu.uuid, date = new Date(99999999), location = "Amsterdam")
-  private val newMenuPerDay2 = MenuPerDay(menuUuid = newMenu.uuid, date = new Date(10000), location = "Rotterdam")
-
-  override def beforeAll {
-    cleanDatabase()
-
-    Await.result(MenuTable.add(newMenu), defaultTimeout)
-  }
-
-  "A MenuPerDay table" must {
-    "add a new menu per day" in {
-      val result = Await.result(MenuPerDayTable.add(newMenuPerDay), defaultTimeout)
-      result.uuid mustBe newMenuPerDay.uuid
-      result.menuUuid mustBe newMenuPerDay.menuUuid
-      result.date.toString mustBe newMenuPerDay.date.toString
-    }
-
-    "query for existing menus per day successfully" in {
-      val result = Await.result(MenuPerDayTable.exists(newMenuPerDay.uuid), defaultTimeout)
-      result mustBe true
-    }
-
-    "query for menus per day by uuid" in {
-      val result = Await.result(MenuPerDayTable.getByUuid(newMenuPerDay.uuid), defaultTimeout)
-      result.map(_.date.toString) mustBe Some(newMenuPerDay).map(_.date.toString)
-    }
-
-    "query for menus per day by menu uuid" in {
-      val result = Await.result(MenuPerDayTable.getByMenuUuid(newMenu.uuid), defaultTimeout)
-      result.map(_.date.toString) mustBe Vector(newMenuPerDay).map(_.date.toString)
-    }
-
-    "query for menus per day by non existent menu uuid" in {
-      val result = Await.result(MenuPerDayTable.getByMenuUuid(UUID.randomUUID()), defaultTimeout)
-      result mustBe Vector()
-    }
-
-    "query for menus per day by date" in {
-      val result = Await.result(MenuPerDayTable.getByDate(new Date(99999999)), defaultTimeout)
-      result.map(_.date.toString) mustBe Vector(newMenuPerDay).map(_.date.toString)
-    }
-
-    "query for menus per day by date that does not exist in table" in {
-      val result = Await.result(MenuPerDayTable.getByDate(new Date(900000000)), defaultTimeout)
-      result mustBe Vector()
-    }
-
-    "query all menus per day" in {
-      val result = Await.result(MenuPerDayTable.getAll, defaultTimeout)
-      result.map(_.date.toString) mustBe Vector(newMenuPerDay).map(_.date.toString)
-    }
-
-    "query all menus per day ordered by date ascending" in {
-      Await.result(MenuPerDayTable.add(newMenuPerDay2), defaultTimeout)
+import shapeless.contrib.scalacheck._
+
+object MenuPerDayTableSpec extends Properties("MenuPerDay") with PropertyTestingConfig {
+
+  import TableDataGenerator._
+
+  override def afterAll(): Unit = dbConnection.db.close()
+
+  private val minDate = 0
+  private val maxDate = 99999999999999L
```

Comment:
Maybe use Long.MaxValue instead? :)

Reply:
No, it was too long. Gave me an exception!

Reply:
Gen.choose gives an exception: `Exception: org.postgresql.util.PSQLException: ERROR: date out of range: "36612232-10-08 +02:00:00`